src/internal_events/kafka.rs

```diff
@@ -140,19 +127,29 @@ impl InternalEvent for KafkaStatisticsReceived<'_> {
     }
 }
 
-pub struct KafkaHeaderExtractionFailed<'a> {
+pub struct KafkaHeaderExtractionError<'a> {
     pub header_field: &'a str,
 }
 
-impl InternalEvent for KafkaHeaderExtractionFailed<'_> {
+impl InternalEvent for KafkaHeaderExtractionError<'_> {
     fn emit_logs(&self) {
         error!(
             message = "Failed to extract header. Value should be a map of String -> Bytes.",
-            header_field = self.header_field
+            error_code = "failed_extracing_header",
+            error_type = error_type::PARSER_FAILED,
+            stage = error_stage::RECEIVING,
+            header_field = self.header_field,
         );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "error_code" => "failed_extracing_field",
+            "error_type" => error_type::PARSER_FAILED,
+            "stage" => error_stage::RECEIVING,
+        );
+        // deprecated
```

Comment:
For comments like this and others, do we have an issue to track that we can link here? Part of the problem with a simple "deprecated" is someone coming a year later to this won't know if we can actually remove this or not.

Reply:
It's not a _huge_ deal but it is helpful if you know of a ticket going. We can generally guess based on how long ago the comment was dropped into the codebase, but if we don't have to guess that's ideal.

Reply:
Yeah, ideally we'd have done this as we went, but I opened https://github.com/vectordotdev/vector/issues/11539 to announce it all at once after all of the component instrumentation work lands.